### DISCUSSION TITLE
Show custom queue channels in Utilities > Queue Manager

### DISCUSSION THREAD

**Discussion by brnquester** [craftcms/cms#13347] [Ideas]
I have some custom queues with different channel names running via systemd worker mostly implemented based on another thread: https://github.com/craftcms/cms/issues/5492 

It works perfectly, no issues. However, I can't see the jobs statuses of those custom queues in the UI of Queue Manager inside the CMS. (`CMS > Utilities > Queue Manager`).

Is there a way to include my custom queues to the Queue Manager view? 

Craft v3.8.9

**Comment by Nellyaa** [comment]
Chiming in to say we're having the same problem with our custom queue channels.
Just spent some hours today trying to implement our custom queues into the Queue Manager, but didn't find a way to do so.
I've come to the solution it isn't possible at the moment.
Feel free to correct me...

**Reply by brandonkelly** [reply]
Yeah it’s not currently possible :(

**Reply by Nellyaa** [reply]
Thanks! Now I know for sure. Hopefully this makes it into Craft at some point :)

**Comment by Wiejeben** [comment]
Would love to see this addition! Right now trying to manage with job priority but it still is one massive pile. For the time being it would be neat if `queue/listen` could have some parameters for minimum and maximum priority, that way I could have a different number of workers available depending on the job priority.

**Reply by Wiejeben** [reply]
Blitz just pushed an option to allow custom queue's, so with that also different channels. Would still really like to see this implemented in the GUI. https://github.com/putyourlightson/craft-blitz/issues/577

**Comment by doit-jochem** [comment]
Hi Everyone and @brandonkelly ,

We've developed our own solution for displaying multiple jobs from various queues, even when each queue has a different database connection.

We extended the queue class from Craft, which is where getJobInfo and getTotalJobs are called from for the job overview in utilities. These functions, as far as I can tell, are used for displaying jobs rather than processing the queue.

Our custom queue class overrides the queue component, ensuring that Craft/Yii always loads our instance when accessing the queue.

Within our custom queue class, we alternate between different queues for the mentioned functions and merge the count and job information to display both sets of queue jobs in the overview:

```php
public function getJobInfo(?int $limit = null): array
{
    $limit        = $limit / 2;
    $defaultTasks = parent::getJobInfo($limit);
    $this->switchConnection('$queueConnectionVar', '$channelVar');
    $syncTasks = parent::getJobInfo($limit);
    $this->switchBack();
    return array_merge($syncTasks, $defaultTasks);
}

private function switchConnection(string|Connection $db, string $channel): void
{
    $this->oldDb      = $this->db;
    $this->oldChannel = $this->channel;
    $this->db         = Instance::ensure($db, Connection::class);
    $this->channel    = $channel;
}

private function switchBack(): void
{
    $this->db      = Instance::ensure($this->oldDb, Connection::class);
    $this->channel = $this->oldChannel;
}

``` 
With this implementation, both queues and their respective jobs are now visible in the utilities control panel.

If anyone has a better suggestion or an improvement on the above, I'd be happy to hear it.

**Reply by wijzijnweb** [reply]
Edit: job details and job actions don't work with the implementation above because they don't switch between queues and think ID 1 (for example) belongs to the primary queue.

**Comment by jornwildenbeest** [comment]
@brandonkelly would love to see this feature. Any updates on this?
